sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/FieldsetWithoutLegendCheck.java

```diff
@@ -40,12 +40,12 @@ public void startElement(TagNode node) {
   @Override
   public void endElement(TagNode node) {
     if (isFieldSet(node)) {
-      if (!foundLegend && fieldsetLine != 0) {
-        createViolation(fieldsetLine, "Add a <legend> tag to this fieldset.");
+      if (!foundLegend && fieldset != null) {
```

Comment:
Is it possible that node.getStartLinePosition() returns 0?
If no, everything is fine, if yes we miss the case where fieldset is not null but equals to zero!

The same happen in FlashUsesBothObjectAndEmbedCheck.java.

Reply:
theoretically, you are right, but the rule doesn't care about the special case of fieldset at line 0, it should detect the fieldset regardless of its position.